core/src/main/java/jenkins/security/s2m/FilePathRuleConfig.java

```diff
@@ -40,7 +40,7 @@ protected FilePathRule parse(String line) {
         if (line.isEmpty())     return null;
 
         line = line.replace("<BUILDDIR>","<JOBDIR>/builds/<BUILDID>");
-        line = line.replace("<BUILDID>","[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]_[0-9][0-9]-[0-9][0-9]-[0-9][0-9]");
+        line = line.replace("<BUILDID>","(?:[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]_[0-9][0-9]-[0-9][0-9]-[0-9][0-9]|[0-9]+)");
```

Comment:
I have no idea what this does - but hasn't the migration changed the buildIDs by now and removed the old timestamped folders from disk?
Also is buildID 0 valid - what about looping - could we have -323453245345345345